components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -71,16 +74,52 @@ public boolean isSigned(RequestObject requestObject) {
     }
 
     @Override
-    public boolean validateSignature(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) throws
-            RequestObjectException {
+    public boolean validateSignature(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) {
 
+        Certificate certificate = null;
+        boolean isVerified = false;
+        String jwksUri = StringUtils.EMPTY;
         SignedJWT jwt = requestObject.getSignedJWT();
-        Certificate certificate =
-                getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
-        boolean isVerified = isSignatureVerified(jwt, certificate);
-        requestObject.setIsSignatureValid(isVerified);
-        return isVerified;
-    }
+        try {
+            certificate =
+                    getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
+        } catch (RequestObjectException e) {
+            log.warn("Public certificate not configured for Service Provider checking whether a jwks endpoint is " +
```

Comment:
Are we sure that this exception is thrown only when certificate is not configured ? If not it's safe to log the stacktrace of exception as a debug log..

Reply:
logged the exception